app/src/ui/changes/commit-message.tsx

```diff
@@ -756,6 +761,7 @@ export class CommitMessage extends React.Component<
             onContextMenu={this.onAutocompletingInputContextMenu}
             disabled={this.props.isCommitting === true}
             spellcheck={this.props.commitSpellcheckEnabled}
+            warningMessage={charactersOverLimit > 0 ? { title:'This commit summary is too long',description: <>The summary is limited to {MaxSummaryLength} characters, which you've exceeded. <strong>{charactersOverLimit} {charactersOverLimit === 1 ? 'character':'characters'}</strong> going over the limit will be cut off.</> }: undefined}
```

Comment:
Thoughts on something like this? We don't want to indicate that it's going to delete something that's over 72 characters, so just trying to be more clear about that.
```suggestion
            warningMessage={charactersOverLimit > 0 ? { title:'This commit summary is longer than recommended',description: <>The summary will be truncated after {MaxSummaryLength} characters, which you've exceeded. <strong>{charactersOverLimit} {charactersOverLimit === 1 ? 'character':'characters'}</strong> going over the limit will be cut off from the summary and wrap into the description.</> }: undefined}
```

Reply:
After looking into the origin of this feature request, I think it'll make more sense to have something like… "It is recommended to keep the commit summary under 50 characters, which you have exceeded by X characters. Learn more.", where "Learn more" is a link to https://git-scm.com/docs/git-commit#_discussion

And then, of course, use a different limit constant of 50 (leave that 72 for UI purposes when long summaries are wrapped into the commit description) to match the official recommendation in Git's documentation.

What do you think?

Reply:
Thanks for clarifying, I see your point. My thoughts on solutions are:
- The popover's title needs to be short enough to fit in without a line break, otherwise it takes up an awkward amount of space. Hence "This commit summary is longer than recommended" isn't great, as it's super verbose and would require at least two lines. I went for "This commit summary is very long" instead – somewhat less authoritative than before, but still really concise.
- It's a great point that the extra characters actually go into the description and don't just disappear. Also wanting to avoid saying "the summary" twice, I went for this: "The summary is limited to 72 characters, which you've exceeded. **n** characters going over the limit will be moved into the commit description." It's also reasonably short for the amount of information, but it is factual: the summary CAN'T be longer than 72 in GitHub Desktop, so it is a hard limit over which the commit description comes into play.
- I think 50 characters is _a_ guideline, but personally I find it debatable and including it in the app would have low impact – it's maybe a nice-to-have. The actually painful thing is summaries getting truncated without warning, so I believe shipping this warning is a lot higher priority than a "nudge" towards 50, I'd rather move forward without it for now.